### PR TITLE
Add support for JsonUnwrapped

### PR DESCRIPTION
**Pull request by tsachev**
Here is PR that adds support for `@JsonUnwrapped` properties that are proxies.

Fix for #97

### CODE REVIEW

**Comment by tsachev** [comment]
I created the PR against 2.11 since this is the current stable branch, but it currently cannot be directly merged in master.

File: hibernate4/src/test/java/com/fasterxml/jackson/datatype/hibernate4/UnwrappedTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+            ObjectMapper mapper = mapperWithModule(true);
+
+            Customer customer = em.find(Customer.class, 500);
+            Product product = customer.getMissingProduct();
+            assertFalse(Hibernate.isInitialized(product));
+
+            String json = mapper.writeValueAsString(new HasUnwrapped<>(product));
+
+            System.out.println(json);
```
Comment (cowtowncoder) [inline hibernate4/src/test/java/com/fasterxml/jackson/datatype/hibernate4/UnwrappedTest.java:51]: Could you please remove this line

**Comment by cowtowncoder** [comment]
First of all: thank you for this contribution!

One quick question: do you think deserialization also works? If so, would be nice to just add code to read serialization back.
If not, that's fine too, just needs to be noted (could file a separate issue for follow-up work).

Beyond that, merging into 2.11 is fine as no new functionality is added, but I suspect that a new feature merged into 2.12 (see #136) might make merging 2.11 -> 2.12 bit difficult.
But I can probably figure it out, and maybe start removing some of intermediate constructors to remove long chaining. So that's fine.

And then the final thing: unless I have asked for (I do not think I have, but apologies if I have) CLA yet, I would need one filled:

https://github.com/FasterXML/jackson/blob/master/contributor-agreement.pdf

this only needs to be done once before the first contribution and then it works for all future PRs.
It is usually easiest to print, fill  & sign, scan/photo, email to `info` at fasterxml dot com.
Once I get that, I can proceed with merging.

Once again, thank you for contributing this fix; looking forward to merging it!

**Comment by cowtowncoder** [comment]
Ah never mind! I noticed you already sent the CLA, thank you very much!

**Comment by tsachev** [comment]
@cowtowncoder thank you for your comments.

I changed the tests so they are now not printing to the console and to read back the serialized json.
It turns out that reading the value works as expected.

Note that I force pushed my changes to keep them in a single commit.

**Comment by cowtowncoder** [comment]
@tsachev Excellent, thank you. I will have one more look before merging.

**Comment by ptahchiev** [comment]
This PR was merged into 2.11 branch, and 2.11.1 was released but I don't see the changes in there:
https://github.com/FasterXML/jackson-datatype-hibernate/blob/jackson-datatype-hibernate-parent-2.11.1/hibernate5/src/main/java/com/fasterxml/jackson/datatype/hibernate5/HibernateProxySerializer.java
Why is that?

**Pull request merged**